src/arkode/arkode_mristep.c

```diff
@@ -441,15 +441,19 @@ int MRIStepReset(void* arkode_mem, realtype tR, N_Vector yR)
                                  &ark_mem, &step_mem);
   if (retval != ARK_SUCCESS) return(retval);
 
-  /* Initialize main ARKode infrastructure */
+  /* Initialize main ARKODE infrastructure */
   retval = arkInit(ark_mem, tR, yR, RESET_INIT);
 
   if (retval != ARK_SUCCESS) {
-    arkProcessError(ark_mem, retval, "ARKode::MRIStep", "MRIStepReset",
-                    "Unable to initialize main ARKode infrastructure");
+    arkProcessError(ark_mem, retval, "ARKODE::MRIStep", "MRIStepReset",
+                    "Unable to initialize main ARKODE infrastructure");
     return(retval);
   }
 
+  /* Reset the inner integrator with this same state */
+  retval = mriStepInnerStepper_Reset(step_mem->stepper, tR, yR);
+  if (retval != ARK_SUCCESS)  return(ARK_INNERSTEP_FAIL);
```

Comment:
The documentation for `MRIStepReset` needs to be updated to note this change in behavior.